[CI] :x-pack:qa:rolling-upgrade tests failing

### CI Link

https://gradle-enterprise.elastic.co/s/kja5p5zfbbroq

### Repro line

 ./gradlew ':x-pack:qa:rolling-upgrade:v8.5.0#bwcTest'

### Does it reproduce?

Didn't try

### Applicable branches

main

### Failure history

_No response_

### Failure excerpt

The root cause is a EOF exception reading a stream input

```
Caused by: java.io.EOFException
	at org.elasticsearch.common.bytes.BytesReferenceStreamInput.moveToNextSlice(BytesReferenceStreamInput.java:195) ~[elasticsearch-8.6.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.bytes.BytesReferenceStreamInput.maybeNextSlice(BytesReferenceStreamInput.java:184) ~[elasticsearch-8.6.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.bytes.BytesReferenceStreamInput.readByte(BytesReferenceStreamInput.java:44) ~[elasticsearch-8.6.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.io.stream.StreamInput.readVIntSlow(StreamInput.java:211) ~[elasticsearch-8.6.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.io.stream.StreamInput.readVInt(StreamInput.java:207) ~[elasticsearch-8.6.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.bytes.BytesReferenceStreamInput.readVInt(BytesReferenceStreamInput.java:113) ~[elasticsearch-8.6.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.io.stream.FilterStreamInput.readVInt(FilterStreamInput.java:60) ~[elasticsearch-8.6.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.unit.ByteSizeUnit.readFrom(ByteSizeUnit.java:307) ~[elasticsearch-8.6.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.unit.ByteSizeValue.<init>(ByteSizeValue.java:70) ~[elasticsearch-8.6.0-SNAPSHOT.jar:?]
	at org.elasticsearch.common.io.stream.StreamInput.readOptionalWriteable(StreamInput.java:992) ~[elasticsearch-8.6.0-SNAPSHOT.jar:?]
	at org.elasticsearch.xpack.core.ml.inference.assignment.AssignmentStats.<init>(AssignmentStats.java:462) ~[?:?]
```

Probably caused by #90464

--- a/build.gradle
+++ b/build.gradle
@@ -137,9 +137,9 @@ tasks.register("verifyVersions") {
  * after the backport of the backcompat code is complete.
  */
 
-boolean bwc_tests_enabled = false
+boolean bwc_tests_enabled = true
 // place a PR link here when committing bwc changes:
-String bwc_tests_disabled_issue = "https://github.com/elastic/elasticsearch/issues/90516"
+String bwc_tests_disabled_issue = ""
 if (bwc_tests_enabled == false) {
   if (bwc_tests_disabled_issue.isEmpty()) {
     throw new GradleException("bwc_tests_disabled_issue must be set when bwc_tests_enabled == false")

